homeassistant/components/blink/switch.py

```diff
@@ -0,0 +1,106 @@
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DEFAULT_BRAND, DOMAIN, TYPE_CAMERA_ARMED
+from .coordinator import BlinkUpdateCoordinator
+
+SWITCH_TYPES: tuple[SwitchEntityDescription, ...] = (
+    SwitchEntityDescription(
+        key=TYPE_CAMERA_ARMED,
+        icon="mdi:motion-sensor",
+        translation_key="camera_motion",
+        device_class=SwitchDeviceClass.SWITCH,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Blink switches."""
+    coordinator: BlinkUpdateCoordinator = hass.data[DOMAIN][config.entry_id]
+
+    entities = [
+        BlinkSwitch(coordinator, camera, description)
+        for camera in coordinator.api.cameras
+        for description in SWITCH_TYPES
+    ]
+    async_add_entities(entities)
+
+
+class BlinkSwitch(CoordinatorEntity[BlinkUpdateCoordinator], SwitchEntity):
+    """Representation of a Blink motion detection switch."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+
+    def __init__(
+        self,
+        coordinator: BlinkUpdateCoordinator,
+        camera,
+        description: SwitchEntityDescription,
+    ) -> None:
+        """Initialize the switch."""
+        super().__init__(coordinator)
+        self._coordinator = coordinator
+        self._prev_state = None
+        self._camera = coordinator.api.cameras[camera]
+        self.entity_description = description
+        serial = self._camera.serial
+        self._attr_unique_id = f"{serial}-{description.key}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(DOMAIN, serial)},
+            serial_number=serial,
+            name=camera,
+            manufacturer=DEFAULT_BRAND,
+            model=self._camera.camera_type,
+        )
+        self._update_attrs()
+
+    async def async_turn_on(self, **kwargs: Any) -> None:
+        """Turn the switch on."""
+        try:
+            await self._camera.async_arm(True)
+
+        except asyncio.TimeoutError as er:
+            raise HomeAssistantError("Blink failed to arm camera") from er
+
+        await self._coordinator.async_refresh()
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the switch off."""
+        try:
+            await self._camera.async_arm(False)
+
+        except asyncio.TimeoutError as er:
+            raise HomeAssistantError("Blink failed to dis-arm camera") from er
+
+        await self._coordinator.async_refresh()
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle update from data coordinator."""
+        self._update_attrs()
+        super()._handle_coordinator_update()
+
+    @callback
+    def _update_attrs(self) -> None:
+        """Update attributes for binary sensor."""
+        self._attr_is_on = self._camera.motion_enabled
```

Comment:
Personal preference: Since you're only updating 1 attribute (and not a whole bunch) maybe this is a cleaner option. But if you think it doesn't fit in with the rest of the platforms or you don't like this, feel free to ignore.
```suggestion
    @property
    def is_on(self) -> bool:
        """Return if the camera is on."""
        return self._camera.motion_enabled
```